packages/injective-testing/src/chain_mock.rs

```diff
@@ -0,0 +1,331 @@
+
+pub struct ExecuteAssertionContainer<Msg> {
+    pub assertion: Option<ExecuteAssertion<Msg>>,
+}
+
+impl<Msg> ExecuteAssertionContainer<Msg> {
+    pub fn new(assertion: ExecuteAssertion<Msg>) -> Self {
+        ExecuteAssertionContainer { assertion: Some(assertion) }
+    }
+
+    pub fn empty() -> Self {
+        ExecuteAssertionContainer { assertion: None }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.assertion.is_none()
+    }
+}
+
+pub struct QueryAssertionContainer<Query> {
+    pub assertion: Option<QueryAssertion<Query>>,
+}
+
+impl<Query> QueryAssertionContainer<Query> {
+    pub fn new(assertion: QueryAssertion<Query>) -> Self {
+        QueryAssertionContainer { assertion: Some(assertion) }
+    }
+
+    pub fn empty() -> Self {
+        QueryAssertionContainer { assertion: None }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.assertion.is_none()
+    }
+}
+
+pub struct CustomInjectiveHandlerAssertions<Msg, Query> {
+    pub executes: Vec<ExecuteAssertionContainer<Msg>>,
+    pub queries: Vec<QueryAssertionContainer<Query>>,
+}
+
+impl<InjectiveMsgWrapper, InjectiveQueryWrapper> Default for CustomInjectiveHandlerAssertions<InjectiveMsgWrapper, InjectiveQueryWrapper> {
+    fn default() -> Self {
+        Self {
+            executes: vec![],
+            queries: vec![],
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct CustomInjectiveHandlerResponses {
+    pub executes: Vec<ExecuteResponseContainer>,
+    pub queries: Vec<QueryResponseContainer>,
+}
+
+#[derive(Default)]
+pub struct CustomInjectiveHandler {
+    pub state: CachingCustomHandlerState<CustomInjectiveHandler, InjectiveMsgWrapper, InjectiveQueryWrapper>,
+    pub responses: CustomInjectiveHandlerResponses,
+    pub assertions: CustomInjectiveHandlerAssertions<InjectiveMsgWrapper, InjectiveQueryWrapper>,
+}
+
+impl Module for CustomInjectiveHandler {
+    type ExecT = InjectiveMsgWrapper;
+    type QueryT = InjectiveQueryWrapper;
+    type SudoT = Empty;
+
+    fn execute<ExecC, QueryC>(
+        &self,
+        _api: &dyn Api,
+        _storage: &mut dyn Storage,
+        _router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        _block: &BlockInfo,
+        _sender: Addr,
+        msg: Self::ExecT,
+    ) -> AnyResult<AppResponse> {
+        let mut exec_calls_count = self.state.execs.borrow().len();
+
+        if !self.assertions.executes.is_empty()
+            && exec_calls_count < self.assertions.executes.len()
+            && !self.assertions.executes[exec_calls_count].is_empty()
+        {
+            self.assertions.executes[exec_calls_count].assertion.unwrap()(&msg);
+        }
+
+        self.state.execs.borrow_mut().push(msg);
+        exec_calls_count += 1;
+
+        if self.responses.executes.is_empty()
+            || exec_calls_count > self.responses.executes.len()
+            || self.responses.executes[exec_calls_count - 1].is_empty()
+        {
+            Ok(AppResponse::default())
```

Comment:
I know people do this often in Rust, but I think it's so much more readable to here instead do

```
    return Ok(AppResponse::default());
}

[normal code without else]
```

Reply:
yeah, but I'll stick to Rust way ;-)

Reply:
Why? I don't think we should follow bad patterns.